runners/spark/src/main/java/org/apache/beam/runners/spark/SparkRunner.java

```diff
@@ -246,17 +258,28 @@ public SparkPipelineResult run(final Pipeline pipeline) {
       result = new SparkPipelineResult.BatchMode(startPipeline, jsc);
     }
 
-    if (mOptions.getEnableSparkMetricSinks()) {
-      registerMetricsSource(mOptions.getAppName());
+    if (pipelineOptions.getEnableSparkMetricSinks()) {
+      registerMetricsSource(pipelineOptions.getAppName());
     }
 
     // it would have been better to create MetricsPusher from runner-core but we need
     // runner-specific
     // MetricsContainerStepMap
     MetricsPusher metricsPusher =
         new MetricsPusher(
-            MetricsAccumulator.getInstance().value(), mOptions.as(MetricsOptions.class), result);
+            MetricsAccumulator.getInstance().value(),
+            pipelineOptions.as(MetricsOptions.class),
+            result);
     metricsPusher.start();
+
+    if (eventLoggingListener != null && jsc != null) {
```

Comment:
I'm curious how `jsc` could be null here?

Reply:
It can't it would have definitely failed before but I am double checking just to be sure.